scalopus_tracing/include/scalopus_tracing/internal/counter_tracepoint.h

```diff
@@ -0,0 +1,46 @@
+/*
+  Copyright (c) 2018-2019, Ivor Wanders
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+
+  * Redistributions of source code must retain the above copyright notice, this
+    list of conditions and the following disclaimer.
+
+  * Redistributions in binary form must reproduce the above copyright notice,
+    this list of conditions and the following disclaimer in the documentation
+    and/or other materials provided with the distribution.
+
+  * Neither the name of the author nor the names of contributors may be used to
+    endorse or promote products derived from this software without specific
+    prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+#ifndef SCALOPUS_TRACING_COUNTER_TRACEPOINT_H
+#define SCALOPUS_TRACING_COUNTER_TRACEPOINT_H
+
+#include <string>
+
+namespace scalopus
+{
+/**
+ * @brief Emits a counter event.
+ * @param id The tracepoint id to relate it to a human readable string.
+ * @param counter_name The counter name.
+ * @param counter_value The counter value.
+ */
+void counter_event(const unsigned int id, const std::string& counter_name, const unsigned int counter_value);
```

Comment:
At first I was going to comment that `counter_name` is superfluous as it should be stored in the `id` via the [StaticStringTracker](https://github.com/iwanders/scalopus/blob/master/scalopus_tracing/include/scalopus_tracing/internal/static_string_tracker.h#L45).

However, then I started experimenting a bit with a trace file. Basically you can have a counter with a name, and in that single counter you can have multiple series. 

See: [trace_with_counters.json.gz](https://github.com/iwanders/scalopus/files/3643607/trace_with_counters.json.gz)
Where I made three counter rows, one with animal counts, one with animal sounds. Both have entries for cats and dogs. I also tested whether negative values are supported. They are in the data format, but they don't get visualised well. I do feel like we should support them. So lets go with `std::int64_t` as value type, realising this, I'll do a cleanup some day to convert all `unsigned int` type definitions to their `cstdint` flavours. No need to do that now.

The main question here is how do we want to represent both the counter name, and the series name. We could put both in the `StaticStringTracker`. That way the signature becomes something like:
```cpp
void counter_event(const unsigned int counter_id, const unsigned int series_id, const std::uint64_t counter_value);
```
Downside here is that ideally the name strings should be compile time constant to leverage the compile time constant crc. This would be the fastest way to deal with these counters, but it makes them less flexible and doesn't really allow us to add or remove series at runtime.

Reply:
`void counter_event(const unsigned int counter_id, const std::int64_t counter_value);`

Reply:
Where `counter_id` would map to a string separated by `/`